computer_vision/haralick_descriptors.py

```diff
@@ -0,0 +1,308 @@
+           [ 0.333333,  0.444444,  1.      ]], dtype=float32)
+    >>> normalize_image(np.array([[4, 4, 3], [1, 7, 2]]))
+
+    """
+    normalized = (image - np.min(image)) / (np.max(image) - np.min(image)) * cap
+    return normalized.astype(np.uint8)
+
+
+def normalize_array(array: np.ndarray, cap: float = 1) -> np.ndarray:
+    """Normalizes a 1D array, between ranges 0-cap.
+
+    Args:
+        array: List containing values to be normalized between cap range.
+        cap: Maximum cap amount for normalization.
+    Returns:
+        return 1D numpy array , corresponding to limited range array
+    """
+    normalized = (array - np.min(array)) / (np.max(array) - np.min(array)) * cap
+    return normalized
+
+
+def grayscale(image: np.ndarray) -> np.ndarray:
+    """Uses luminance weights to transform RGB channel to greyscale, by
+    taking the dot product between the channel and the weights."""
+    return np.dot(image[:, :, 0:3], [0.299, 0.587, 0.114]).astype(np.uint8)
+
+
+def binarize(image: np.ndarray, threshold_value) -> np.ndarray:
+    """
+    Binarizes a grayscale image based on a given threshold value,
+    setting values to 1 or 0 accordingly.
+    """
+    binarized = np.where(image > threshold_value, 1, 0)
+
+    return binarized
+
+
+def transform(image: np.ndarray, kind, kernel=None) -> np.ndarray:
+    """
+    Simple image transformation using one of two available filter functions:
+    Erosion and Dilation.
+
+    Args:
+        image:
+        kind: Can be either 'erosion', in which case the :func:np.max
+              function is called, or 'dilation', when :func:np.min is used instead.
+        kernel: n x n kernel with shape < :attr:image.shape,
+              to be used when applying convolution to original image
+
+    Returns:
+
+    """
+    if kernel is None:
+        kernel = np.ones((3, 3))
+
+    if kind == "erosion":
+        constant = 1
+        apply = np.max
+    else:
+        constant = 0
+        apply = np.min
+
+    center_x, center_y = (x // 2 for x in kernel.shape)
+
+    # Use padded image when applying convolotion
+    # to not go out of bounds of the original the image
+    transformed = np.zeros(image.shape, dtype=np.uint8)
+    padded = np.pad(image, 1, "constant", constant_values=constant)
+
+    for x in range(center_x, padded.shape[0] - center_x):
+        for y in range(center_y, padded.shape[1] - center_y):
+            center = padded[
+                x - center_x : x + center_x + 1, y - center_y : y + center_y + 1
+            ]
+            # Apply transformation method to the centered section of the image
+            transformed[x - center_x, y - center_y] = apply(center[kernel == 1])
+
+    return transformed
+
+
+def opening_filter(image: np.ndarray, kernel: np.ndarray = None) -> np.ndarray:
+    """
+    Opening filter, defined as the sequence of
+    erosion and then a dilation filter on the same image.
+    """
+    if kernel is None:
```

Comment:
In this instance, all falsey values should be treated the same way as None.
```suggestion
    if not kernel:
```